### DISCUSSION TITLE
How to make HTTP requests with Javascript in Nuclei

### DISCUSSION THREAD

**Discussion by TimMaasGeesteranus** [projectdiscovery/nuclei#5741] [Q&A]
I was wondering whether it is possible to make HTTP requests in JavaScript. There seems to be no module that supports this. 

My usecase: 
I would like to load endpoints with parameters from one file, fill in the parameters with payloads from another file, and use the results as payloads for my HTTP requests. I currently don't see any other way of doing this than using javascript

It would also help if the updatePayload function would upload the payloads in the HTTP request

```
id: exploit

info:
  name: test
  author: me
  severity: critical
  description: test

javascript:

    payloads:
      path: 
        - "/test"
        - "/anothertest"      

    code: |
      const paths = fs.ReadFileAsString('paths.txt').split('\n');
      updatePayload('path', paths)

      let m = require('nuclei/net');
      let address = "https://some-random-api.com";
      let host = "some-random-api.com"
      let conn = m.OpenTLS('tcp', address);
      conn.Send('GET {{path}} HTTP/2\r\nHost:'+host+'\r\nConnection: close\r\n\r\n');
      conn.RecvString();
```

**Comment by dogancanbakir** [comment]
This discussion closed automatically due to inactivity. Feel free to reopen or start new if still relevant.
